src/Voron/Impl/SliceSmallSet.cs

```diff
@@ -60,38 +66,36 @@ private IEnumerable<TValue> ReturnValues()
 
         public unsafe void Add(Slice key, TValue value)
         {
-            if (_overflowStorage != null)
-            {
-                goto Overflow;
-            }
-
             int idx = FindKey(key);
-            if (idx == Invalid)
+            if (idx != Invalid)
+                goto Done;
+
+            // side effect, overflow if needed
+            idx = RequestWritableBucket();
+            if (idx == Invalid || _overflowStorage != null)
             {
-                // side affect, overflow if needed
-                idx = RequestWritableBucket();
-                if (idx == Invalid)
-                    goto Overflow;
+                Debug.Assert(_overflowStorage != null);
+                _overflowStorage[key] = value;
             }
 
+            Done:
             _keys[idx] = key;
             _keySizes[idx] = key.Size;
             _keyHashes[idx] = Hashing.XXHash64.CalculateInline(key.Content.Ptr, (ulong)key.Size);
             _values[idx] = value;
```

Comment:
Why are you still using those if you are in overflow?

Reply:
Yes. See other comment.